pkg/kubectl/describe.go

```diff
@@ -53,7 +53,13 @@ import (
 // if the output could not be generated. Implementers typically
 // abstract the retrieval of the named object from a remote server.
 type Describer interface {
-	Describe(namespace, name string) (output string, err error)
+	Describe(namespace, name string, describerSetting DescriberSetting) (output string, err error)
+}
+
+// DescriberSetting holds display configuration for each object
+// describer to control what is printed.
+type DescriberSetting struct {
```

Comment:
I though setting is uncountable till this moment. Looks like google says "use plural if there are more things to set". ``DescriberConfig`` would work too. If the describer is meant just to "render", rendering settings is quite common. I like this comment [1]. ``DescriberPreferences`` is the closest to the behaviour from the point of the describer. On the other hand ``DescriberOptions`` makes more sense from the point of the ``kubectl describe`` command as ``events=true`` is by default and changing to false is providing option to change it.

``DescribeOptions`` is already use so ``DescriberOptions`` counts out. Let's use ``DescriberSettings`` if noone against :).

[1] http://stackoverflow.com/questions/2074384/options-settings-properties-configuration-preferences-when-and-why/28812395#28812395